### ISSUE TITLE
test_fence_hints is flaky

### ISSUE THREAD

**Issue by xemul** [scylladb/scylladb#18012]
Caught while testing changes in test.py script (#17981)
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7693/

```
[2024-03-25T17:22:15.764Z] =================================== FAILURES ===================================
[2024-03-25T17:22:15.764Z] _______________________________ test_fence_hints _______________________________
[2024-03-25T17:22:15.764Z] 
[2024-03-25T17:22:15.764Z] self = <aiohttp.client_proto.ResponseHandler object at 0x7f2ec817c050>
[2024-03-25T17:22:15.764Z] exc = ClientPayloadError('Response payload is not completed')
[2024-03-25T17:22:15.764Z] 
[2024-03-25T17:22:15.764Z]     def connection_lost(self, exc: Optional[BaseException]) -> None:
[2024-03-25T17:22:15.764Z]         self._drop_timeout()
[2024-03-25T17:22:15.764Z]     
[2024-03-25T17:22:15.764Z]         if self._payload_parser is not None:
[2024-03-25T17:22:15.764Z]             with suppress(Exception):
[2024-03-25T17:22:15.764Z]                 self._payload_parser.feed_eof()
[2024-03-25T17:22:15.764Z]     
[2024-03-25T17:22:15.764Z]         uncompleted = None
[2024-03-25T17:22:15.764Z]         if self._parser is not None:
[2024-03-25T17:22:15.764Z]             try:
[2024-03-25T17:22:15.764Z] >               uncompleted = self._parser.feed_eof()
[2024-03-25T17:22:15.764Z] 
[2024-03-25T17:22:15.764Z] /usr/lib64/python3.11/site-packages/aiohttp/client_proto.py:83: 
[2024-03-25T17:22:15.764Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2024-03-25T17:22:15.764Z] 
[2024-03-25T17:22:15.764Z] >   ???
[2024-03-25T17:22:15.764Z] E   aiohttp.http_exceptions.TransferEncodingError: 400, message:
[2024-03-25T17:22:15.764Z] E     Not enough data for satisfy transfer length header.
[2024-03-25T17:22:15.764Z] 
[2024-03-25T17:22:15.764Z] aiohttp/_http_parser.pyx:507: TransferEncodingError
[2024-03-25T17:22:15.764Z] 
[2024-03-25T17:22:15.764Z] The above exception was the direct cause of the following exception:
[2024-03-25T17:22:15.765Z] 
[2024-03-25T17:22:15.765Z] request = <FixtureRequest for <Function test_fence_hints>>
[2024-03-25T17:22:15.765Z] manager = <test.pylib.manager_client.ManagerClient object at 0x7f2ec9b680d0>
[2024-03-25T17:22:15.765Z] 
[2024-03-25T17:22:15.765Z]     @pytest.mark.asyncio
[2024-03-25T17:22:15.765Z]     @skip_mode('release', 'error injections are not supported in release mode')
[2024-03-25T17:22:15.765Z]     async def test_fence_hints(request, manager: ManagerClient):
[2024-03-25T17:22:15.765Z]         logger.info("Bootstrapping cluster with three nodes")
[2024-03-25T17:22:15.765Z]         s0 = await manager.server_add(config={
[2024-03-25T17:22:15.765Z]             'error_injections_at_startup': ['decrease_hints_flush_period']
[2024-03-25T17:22:15.765Z]         }, cmdline=['--logger-log-level', 'hints_manager=trace'])
[2024-03-25T17:22:15.765Z]     
[2024-03-25T17:22:15.765Z]         # Disable load balancer as it might bump topology version, potentially creating a race condition
[2024-03-25T17:22:15.765Z]         # with read modify write below.
[2024-03-25T17:22:15.765Z]         # This should be done before adding the last two servers,
[2024-03-25T17:22:15.765Z]         # otherwise it can break the version == fence_version condition
[2024-03-25T17:22:15.765Z]         # which the test relies on.
[2024-03-25T17:22:15.765Z]         await manager.api.disable_tablet_balancing(s0.ip_addr)
[2024-03-25T17:22:15.765Z]     
[2024-03-25T17:22:15.765Z]         [s1, s2] = await manager.servers_add(2)
[2024-03-25T17:22:15.765Z]     
[2024-03-25T17:22:15.765Z]         logger.info(f'Creating test table')
[2024-03-25T17:22:15.765Z]         random_tables = RandomTables(request.node.name, manager, unique_name(), 3)
[2024-03-25T17:22:15.765Z]         table1 = await random_tables.add_table(name='t1', pks=1, columns=[
[2024-03-25T17:22:15.765Z]             Column("pk", IntType),
[2024-03-25T17:22:15.765Z]             Column('int_c', IntType)
[2024-03-25T17:22:15.765Z]         ])
[2024-03-25T17:22:15.765Z]         cql = manager.get_cql()
[2024-03-25T17:22:15.765Z]         await cql.run_async(f"USE {random_tables.keyspace}")
[2024-03-25T17:22:15.765Z]     
[2024-03-25T17:22:15.765Z]         logger.info(f'Waiting for cql and hosts')
[2024-03-25T17:22:15.765Z]         hosts = await wait_for_cql_and_get_hosts(cql, [s0, s2], time.time() + 60)
[2024-03-25T17:22:15.765Z]     
[2024-03-25T17:22:15.765Z]         host2 = host_by_server(hosts, s2)
[2024-03-25T17:22:15.765Z]         new_version = (await get_version(manager, host2)) + 1
[2024-03-25T17:22:15.765Z]         logger.info(f"Set version and fence_version to {new_version} on node {host2}")
[2024-03-25T17:22:15.765Z]         await set_version(manager, host2, new_version)
[2024-03-25T17:22:15.766Z]         await set_fence_version(manager, host2, new_version)
[2024-03-25T17:22:15.766Z]     
[2024-03-25T17:22:15.766Z]         select_all_stmt = SimpleStatement("select * from t1", consistency_level=ConsistencyLevel.ONE)
[2024-03-25T17:22:15.766Z]         rows = await cql.run_async(select_all_stmt, host=host2)
[2024-03-25T17:22:15.766Z]         assert len(list(rows)) == 0
[2024-03-25T17:22:15.766Z]     
[2024-03-25T17:22:15.766Z]         logger.info(f"Stopping node {host2}")
[2024-03-25T17:22:15.766Z]         await manager.server_stop_gracefully(s2.server_id)
[2024-03-25T17:22:15.766Z]     
[2024-03-25T17:22:15.766Z]         host0 = host_by_server(hosts, s0)
[2024-03-25T17:22:15.766Z]         logger.info(f"Writing through {host0} to regular column")
[2024-03-25T17:22:15.766Z]         await cql.run_async("insert into t1(pk, int_c) values (1, 1)", host=host0)
[2024-03-25T17:22:15.766Z]     
[2024-03-25T17:22:15.766Z]         logger.info(f"Starting last node {host2}")
[2024-03-25T17:22:15.766Z]         await manager.server_start(s2.server_id)
[2024-03-25T17:22:15.766Z]     
[2024-03-25T17:22:15.766Z]         logger.info(f"Waiting for failed hints on {host0}")
[2024-03-25T17:22:15.766Z]         async def at_least_one_hint_failed():
[2024-03-25T17:22:15.766Z]             metrics_data = await manager.metrics.query(s0.ip_addr)
[2024-03-25T17:22:15.766Z]             if sent_metric(metrics_data) > 0:
[2024-03-25T17:22:15.766Z]                 pytest.fail(f"Unexpected successful hints; metrics on {s0}: {all_hints_metrics(metrics_data)}")
[2024-03-25T17:22:15.766Z]             if send_errors_metric(metrics_data) >= 1:
[2024-03-25T17:22:15.766Z]                 return True
[2024-03-25T17:22:15.766Z]             logger.info(f"Metrics on {s0}: {all_hints_metrics(metrics_data)}")
[2024-03-25T17:22:15.766Z] >       await wait_for(at_least_one_hint_failed, time.time() + 60)
[2024-03-25T17:22:15.766Z] 
[2024-03-25T17:22:15.766Z] test/topology_experimental_raft/test_fencing.py:174: 
[2024-03-25T17:22:15.766Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2024-03-25T17:22:15.766Z] test/pylib/util.py:49: in wait_for
[2024-03-25T17:22:15.766Z]     res = await pred()
[2024-03-25T17:22:15.766Z] test/topology_experimental_raft/test_fencing.py:168: in at_least_one_hint_failed
[2024-03-25T17:22:15.766Z]     metrics_data = await manager.metrics.query(s0.ip_addr)
[2024-03-25T17:22:15.766Z] test/pylib/rest_client.py:412: in query
[2024-03-25T17:22:15.766Z]     data = await self.client.get_text('/metrics', host=server_ip)
[2024-03-25T17:22:15.766Z] test/pylib/rest_client.py:82: in get_text
[2024-03-25T17:22:15.766Z]     ret = await self._fetch("GET", resource_uri, response_type = "text", host = host,
[2024-03-25T17:22:15.766Z] test/pylib/rest_client.py:72: in _fetch
[2024-03-25T17:22:15.767Z]     return await getattr(resp, response_type)()
[2024-03-25T17:22:15.767Z] /usr/lib64/python3.11/site-packages/aiohttp/client_reqrep.py:1143: in text
[2024-03-25T17:22:15.767Z]     await self.read()
[2024-03-25T17:22:15.767Z] /usr/lib64/python3.11/site-packages/aiohttp/client_reqrep.py:1101: in read
[2024-03-25T17:22:15.767Z]     self._body = await self.content.read()
[2024-03-25T17:22:15.767Z] /usr/lib64/python3.11/site-packages/aiohttp/streams.py:373: in read
[2024-03-25T17:22:15.767Z]     block = await self.readany()
[2024-03-25T17:22:15.767Z] /usr/lib64/python3.11/site-packages/aiohttp/streams.py:395: in readany
[2024-03-25T17:22:15.767Z]     await self._wait("readany")
[2024-03-25T17:22:15.767Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2024-03-25T17:22:15.767Z] 
[2024-03-25T17:22:15.767Z] self = <StreamReader e=ClientPayloadError('Response payload is not completed')>
[2024-03-25T17:22:15.767Z] func_name = 'readany'
[2024-03-25T17:22:15.767Z] 
[2024-03-25T17:22:15.767Z]     async def _wait(self, func_name: str) -> None:
[2024-03-25T17:22:15.767Z]         # StreamReader uses a future to link the protocol feed_data() method
[2024-03-25T17:22:15.767Z]         # to a read coroutine. Running two read coroutines at the same time
[2024-03-25T17:22:15.767Z]         # would have an unexpected behaviour. It would not possible to know
[2024-03-25T17:22:15.767Z]         # which coroutine would get the next data.
[2024-03-25T17:22:15.767Z]         if self._waiter is not None:
[2024-03-25T17:22:15.767Z]             raise RuntimeError(
[2024-03-25T17:22:15.767Z]                 "%s() called while another coroutine is "
[2024-03-25T17:22:15.767Z]                 "already waiting for incoming data" % func_name
[2024-03-25T17:22:15.767Z]             )
[2024-03-25T17:22:15.767Z]     
[2024-03-25T17:22:15.767Z]         waiter = self._waiter = self._loop.create_future()
[2024-03-25T17:22:15.767Z]         try:
[2024-03-25T17:22:15.768Z]             with self._timer:
[2024-03-25T17:22:15.768Z] >               await waiter
[2024-03-25T17:22:15.768Z] E               aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed
[2024-03-25T17:22:15.768Z] 
[2024-03-25T17:22:15.768Z] /usr/lib64/python3.11/site-packages/aiohttp/streams.py:302: ClientPayloadError
```

**Comment by Annamikhlin** [comment]
seen also here:
https://jenkins.scylladb.com/job/scylla-master/job/next/7500/testReport/junit/(root)/test_fencing/Build___x86___Unit_Tests_x86___test_fence_hints_2/

**Comment by mykaul** [comment]
@kbr-scylla - does that belong to your team?

**Comment by kbr-scylla** [comment]
Duplicate of https://github.com/scylladb/scylladb/issues/18083
